algebra-core/src/curves/glv.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{biginteger::BigInteger, ModelParameters, PrimeField};
+use core::ops::Neg;
+
+/// TODO: deal with the case where b1 and b2 have the same sign
+pub trait GLVParameters: Send + Sync + 'static + ModelParameters {
+    type WideBigInt: BigInteger;
+
+    const LAMBDA: Self::ScalarField; // lambda in ZZ s.t. phi(P) = lambda*P for all P
+    const OMEGA: Self::BaseField; // phi((x, y)) = (\omega x, y)
+    const Q1: <Self::ScalarField as PrimeField>::BigInt; // round(R*|b2|/n)
+    const Q2: <Self::ScalarField as PrimeField>::BigInt; // round(R*|b1|/n)
+    const B1: <Self::ScalarField as PrimeField>::BigInt; // |b1|
+    const B2: <Self::ScalarField as PrimeField>::BigInt; // |b2|
+    const B1_IS_NEG: bool;
+    const B2_IS_NEG: bool;
+    const R_BITS: u32;
+
+    #[inline]
+    fn glv_scalar_decomposition_inner(
+        k: <Self::ScalarField as PrimeField>::BigInt,
+    ) -> (
+        (bool, <Self::ScalarField as PrimeField>::BigInt),
+        (bool, <Self::ScalarField as PrimeField>::BigInt),
+    ) {
+        let limbs = <Self::ScalarField as PrimeField>::BigInt::NUM_LIMBS;
+        let modulus = Self::ScalarField::modulus();
+
+        // If we are doing a subgroup check, we should multiply by the original scalar
+        // since the GLV decomposition does not guarantee that we would not be
+        // adding and subtracting back to zero
+        if k == modulus {
```

Comment:
Yes, I am aware. It should also be possible to handle this case by using biginteger rather than field. I will look into whether it can be handled generically.